sys/include/fido2/ctap/transport/ctap_transport.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2021 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    fido2_ctap_transport CTAP transport layer
+ * @ingroup     fido2_ctap
+ * @brief       CTAP transport layer
+ *
+ * @{
+ *
+ * @file
+ * @brief       CTAP2 transport layer defines and function declarations
+ *
+ * @author      Nils Ollrogge <nils-ollrogge@outlook.de>
+ */
+
+#ifndef FIDO2_CTAP_TRANSPORT_CTAP_TRANSPORT_H
+#define FIDO2_CTAP_TRANSPORT_CTAP_TRANSPORT_H
+
+#include <stdint.h>
+#include "mutex.h"
+#include "timex.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief CTAP supported transport prototocols
+ */
+typedef enum {
+    USB,
+    NFC,
+    BLE
+} ctap_transport_type_t;
+
+/**
+ * @brief CTAP_TRANSPORT thread stack size
+ */
+#define CTAP_TRANSPORT_STACKSIZE 15000
```

Comment:
It needs to be configurable because the implementation should be compatible with different platforms (in future) that might compile a bit differently and eventually require more or less stack. As a user, you want to be able to configure this (via Kconfig) and not go to the header file to change a value.